src/filters/capture.rs

```diff
@@ -14,78 +14,80 @@
  * limitations under the License.
  */
 
-mod capture;
+mod affix;
 mod config;
 mod metrics;
-mod proto;
+mod regex;
 
-use std::sync::Arc;
+crate::include_proto!("quilkin.extensions.filters.capture.v1alpha1");
 
-use tracing::warn;
+use std::sync::Arc;
 
 use crate::{filters::prelude::*, metadata::Value};
 
-use capture::Capture;
-use metrics::Metrics;
-use proto::quilkin::extensions::filters::capture_bytes::v1alpha1::CaptureBytes as ProtoConfig;
+use self::{
+    affix::{Prefix, Suffix},
+    metrics::Metrics,
+    regex::Regex,
+};
 
+use self::quilkin::extensions::filters::capture::v1alpha1 as proto;
 pub use config::{Config, Strategy};
 
-pub const NAME: &str = "quilkin.extensions.filters.capture_bytes.v1alpha1.CaptureBytes";
+pub const NAME: &str = "quilkin.extensions.filters.capture.v1alpha1.Capture";
 
 /// Creates a new factory for generating capture filters.
 pub fn factory() -> DynFilterFactory {
     Box::from(CaptureBytesFactory::new())
 }
 
-struct CaptureBytes {
-    capture: Box<dyn Capture + Sync + Send>,
+/// Trait to implement different strategies for capturing packet data.
+pub trait CaptureStrategy {
+    /// Capture packet data from the contents, and optionally returns a value if
+    /// anything was captured.
+    fn capture(&self, contents: &mut Vec<u8>, metrics: &Metrics) -> Option<Value>;
+}
+
+struct Capture {
+    capture: Box<dyn CaptureStrategy + Sync + Send>,
     /// metrics reporter for this filter.
     metrics: Metrics,
     metadata_key: Arc<String>,
-    size: usize,
-    remove: bool,
+    is_present_key: Arc<String>,
 }
 
-impl CaptureBytes {
+impl Capture {
     fn new(config: Config, metrics: Metrics) -> Self {
-        CaptureBytes {
-            capture: config.strategy.as_capture(),
+        Self {
+            capture: config.strategy.into_capture(),
             metrics,
+            is_present_key: Arc::new(config.metadata_key.clone() + "/is_present"),
```

Comment:
Just noticed that this key is not documented, so we should add that to `capture.md`.

Reply:
This is now documented in the config struct, once #478 is merged it will be documented in `capture.md`.